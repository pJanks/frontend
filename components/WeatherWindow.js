import * as React from 'react';
import { Text, View, Button, TextInput, StyleSheet } from 'react-native';

export default function WeatherWindow() {
  return (
    <View style={ styles.mainContainer }>
      <View style={ styles.tempContainer }>
        <Text style={ styles.highLow }>High: 73° Low: 45°</Text>
        <Text style={ styles.mainTemp }>65°</Text>
      </View>
      <View>
        <Text>[Weather Icon]</Text>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: 'lightyellow',
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  tempContainer: {
    flexDirection: 'column',
  },
  mainTemp: {
    fontSize: 64,
  },
  highLow: {
    fontSize: 12,
  }
})