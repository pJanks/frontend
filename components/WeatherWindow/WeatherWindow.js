import * as React from 'react';
import { Text, View, StyleSheet, Image } from 'react-native';

export default function WeatherWindow({ activity }) {
  const iconDictionary = {
    'clear-day': require('../../assets/clear-day.png'),
    'clear-night': require('../../assets/clear-night.png'),
    'cloudy': require('../../assets/cloudy.png'),
    'fog': require('../../assets/fog.png'),
    'partly-cloudy-day': require('../../assets/partly-cloudy-day.png'),
    'partly-cloudy-night': require('../../assets/partly-cloudy-night.png'),
    'rain': require('../../assets/rain.png'),
    'sleet': require('../../assets/sleet.png'),
    'snow': require('../../assets/snow.png'),
    'wind': require('../../assets/wind.png')
  }

  const icon = iconDictionary[activity.forecast_img];

  return (
    <View style={ styles.mainContainer }>
      <View style={ styles.tempContainer }>
        <Text style={ styles.highLow }>High: { activity.temp_hi }&deg;F Low: { activity.temp_low }&deg;F</Text>
        <Text style={ styles.mainTemp }>{ activity.temperature }&deg;F</Text>
      </View>
      <View>
        <Image 
          style={ styles.forecastImg }
          source={icon} />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: 'lightyellow',
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  tempContainer: {
    flexDirection: 'column',
  },
  mainTemp: {
    fontSize: 64,
  },
  highLow: {
    fontSize: 12,
  },
  forecastImg: {
    padding: 15,
    width: 100,
    height: 100
  }
})
