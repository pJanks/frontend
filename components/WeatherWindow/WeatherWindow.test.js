import React from 'react';
import WeatherWindow from './WeatherWindow';

import renderer from 'react-test-renderer';

test('renders correctly', () => {

  const weatherWindowComponent = renderer.create(<WeatherWindow />).toJSON();

  expect(weatherWindowComponent).toMatchSnapshot();
});
