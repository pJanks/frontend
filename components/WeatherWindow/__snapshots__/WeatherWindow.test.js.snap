// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightyellow",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      High: 73° Low: 45°
    </Text>
    <Text
      style={
        Object {
          "fontSize": 64,
        }
      }
    >
      65°
    </Text>
  </View>
  <View>
    <div
      data-testid="skycon"
    >
      mock skycon
    </div>
  </View>
</View>
`;
